### PR TITLE
[chainerx] add swapaxes

### PR DESCRIPTION
**Pull request by kshitij12345**
References : #6423

### CODE REVIEW

**Comment by hvy** [comment]
Thanks a lot for this PR. Could you in this PR also include the `chainerx.ndarray.swapaxes` method in addition to the module function?

File: chainerx/_docs/routines.py [hunk 1]
```diff
@@ -837,6 +837,28 @@ def _docs_manipulation():
...
+Args:
+    ary (~chainerx.ndarray): Array to swapaxes.
+    axis1 (int): First Axis
+    axis2 (int): Second Axis
+
+Returns:
+    ~chainerx.ndarray: Swaped array.
+
+Note:
+    * Output array is a view of Input array.
```
Comment (hvy) [inline chainerx/_docs/routines.py]: A nitpick.
```suggestion
    * Output array is a view of the input array.
```

File: chainerx/_docs/routines.py [hunk 2]
```diff
@@ -837,6 +837,28 @@ def _docs_manipulation():
...
+    * During backpropagation, this function propagates the gradients of the
+      output arrays to the input array ``ary``.
```
Comment (hvy) [inline chainerx/_docs/routines.py]:
```suggestion
      output arrays to the input array ``a``.
```

File: chainerx_cc/chainerx/routines/manipulation.cc [hunk 3]
```diff
@@ -695,4 +695,35 @@ std::vector<Array> Split(const Array& ary, std::vector<int64_t> indices, int8_t
     return out;
 }
 
+Array Swapaxes(const Array& ary, int8_t axis1, int8_t axis2) {
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/manipulation.cc]: Ditto `a`.
```suggestion
Array Swapaxes(const Array& a, int8_t axis1, int8_t axis2) {
```

File: chainerx_cc/chainerx/routines/manipulation.cc [hunk 4]
```diff
@@ -695,4 +695,35 @@ std::vector<Array> Split(const Array& ary, std::vector<int64_t> indices, int8_t
...
+    Shape shape = ary.shape();
+    Strides strides = ary.strides();
+
+    // for negative axis.
+    axis1 = axis1 >= 0 ? axis1 : ary.ndim() + axis1;
+    axis2 = axis2 >= 0 ? axis2 : ary.ndim() + axis2;
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/manipulation.cc]: We have a function that does exactly this and the error checking https://github.com/chainer/chainer/blob/master/chainerx_cc/chainerx/axes.h#L85.

File: chainerx_cc/chainerx/routines/manipulation.h [hunk 5]
```diff
@@ -49,4 +49,6 @@ std::vector<Array> Split(const Array& ary, int64_t sections, int8_t axis = 0);
 // Returns a set of arrays resulting from splitting the given array at the indices along the specified axis.
 std::vector<Array> Split(const Array& ary, std::vector<int64_t> indices, int8_t axis = 0);
 
+Array Swapaxes(const Array& ary, int8_t axis1, int8_t axis2);
```
Comment (hvy) [inline chainerx_cc/chainerx/routines/manipulation.h]: One last occurrence of `ary`.
```suggestion
Array Swapaxes(const Array& a, int8_t axis1, int8_t axis2);
```

Comment (kshitij12345) [reply chainerx_cc/chainerx/routines/manipulation.h]: Hahah thanks! Nothing slips your eyes.

**Comment by hvy** [comment]
jenkins, test this please.

File: chainerx_cc/chainerx/routines/manipulation.h (hunk 5 above)
Comment (hvy) [reply chainerx_cc/chainerx/routines/manipulation.h]: Quick fix as always, thanks a lot :)

**Comment by kshitij12345** [comment]
Will have to wait till #6922

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
Renamed the PR title for style and release note rendering.

**Pull request merged**